avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -95,10 +99,42 @@ public DatumReader<?> load(IdentityPair<Schema, Schema> key) {
    * Sets properties for this deserializer without overriding the schema registry client itself.
    * Useful for testing, where a mock client is injected.
    */
-  protected void configure(KafkaAvroDeserializerConfig config) {
+  protected void configure(KafkaAvroDeserializerConfig config, boolean isKey) {
     configureClientProperties(config, new AvroSchemaProvider());
     useSpecificAvroReader = config
         .getBoolean(KafkaAvroDeserializerConfig.SPECIFIC_AVRO_READER_CONFIG);
+
+    final String specificAvroClassLookupKey = isKey
+        ? KafkaAvroDeserializerConfig.SPECIFIC_AVRO_KEY_TYPE_CONFIG :
+          KafkaAvroDeserializerConfig.SPECIFIC_AVRO_VALUE_TYPE_CONFIG;
+
+    final Class clazz = config.getClass(specificAvroClassLookupKey);
+    if (useSpecificAvroReader && clazz != null) {
```

Comment:
Per @OneCricketeer , perhaps we just emit a warning if `useSpecificAvroReader` is false.